recipes/install.rb

```diff
@@ -696,8 +697,10 @@
   owner node['glassfish']['user']
   group node['glassfish']['group']
   mode "550"
-  variables({
-              :dtrx => dtrx
+  variables(lazy {
```

Comment:
This is not needed - the previous code was ok.

Reply:
No the previous code was not ok. If it was, we would see the empty string only on centos, but we see it empty also on ubuntu. The reason si that the value of the variable `dtrx` is only known at runtime, while the variables block is (without the lazy) evaluated at the beginning (when dtrx is still emtpy). See: https://docs.chef.io/resource_common.html#lazy-evaluation and

Reply:
ok, sloppy by me, due to sloppy variable scoping in ruby/chef. dtrx should be defined before the platform case statement. The 'variables{lazy{' is not the simplest solution.

This should be the solution - first define the dtrx variable in the recipe scope, then override it in the rhel/centos case statement:

dtrx="dtrx"
switch platform_family:
case "debian":
# do nothing here
case "rhel":
dtrx="/usr/local/bin/dtrx"